### PR TITLE
Fix RPC semantic conventions

### PR DESCRIPTION
**Pull request by arminru**
The previously defined requirement of always providing `net.peer.name` only works for a client connecting to a server.
A server on the other hand is usually not able to provide the host name of a client connecting to it, since it only has access to the IP address of the incoming request.

### CODE REVIEW

File: specification/data-rpc.md [hunk 1]
```diff
@@ -35,7 +35,8 @@ Examples of span name: `grpc.test.EchoService/Echo`.
 | `component`    | Declares that this is a grpc component. Value MUST be `"grpc"` | Yes       |
 | `rpc.service`  | The service name, must be equal to the $service part in the span name. | Yes |
 
-Additionally, the `net.peer.name` and `net.peer.port` [network attributes][] are required.
+Additionally, at least one of `net.peer.name` or `net.peer.ip` is required as well as `net.peer.port`.
+These are defined in [network attributes][].
```
Comment (jmacd) [inline specification/data-rpc.md]: nit: Could you just add a line for `net.peer.port` to the table above, then remove "as well as `net.peer.port`"?

Comment (arminru) [reply specification/data-rpc.md]: sure, done!

File: specification/data-rpc.md [hunk 2]
```diff
@@ -34,8 +34,10 @@ Examples of span name: `grpc.test.EchoService/Echo`.
 | -------------- | ------------------------------------------------------------ | --------- |
 | `component`    | Declares that this is a grpc component. Value MUST be `"grpc"` | Yes       |
 | `rpc.service`  | The service name, must be equal to the $service part in the span name. | Yes |
+| `net.peer.port`  | see [network attributes][] | Yes |
```
Comment (Oberon00) [inline specification/data-rpc.md]:
```suggestion
| `net.peer.port`  | See [network attributes][] | Yes |
```

File: specification/data-rpc.md [hunk 3]
```diff
@@ -34,8 +34,10 @@ Examples of span name: `grpc.test.EchoService/Echo`.
...
 
-Additionally, the `net.peer.name` and `net.peer.port` [network attributes][] are required.
+Additionally, at least one of `net.peer.name` or `net.peer.ip` is required.
+These attributes are also defined in [network attributes][].
```
Comment (Oberon00) [inline specification/data-rpc.md]: Why "also"? As far as I can see, they are *only* defined there.

Comment (arminru) [reply specification/data-rpc.md]: Also, since the `net.peer.port` mentioned in the line above is also defined there.

Comment (arminru) [reply specification/data-rpc.md]: I rephrased it :)

**Comment by tedsuo** [comment]
@arminru approved! If you can sync your branch with master, we can merge this.

**Comment by arminru** [comment]
@tedsuo done! :shipit:

File: specification/data-rpc.md [hunk 4]
```diff
@@ -32,10 +32,11 @@ Examples of span name: `grpc.test.EchoService/Echo`.
 
 | Attribute name | Notes and examples                                           | Required? |
 | -------------- | ------------------------------------------------------------ | --------- |
-| `component`    | Declares that this is a grpc component. Value MUST be `"grpc"` | Yes       |
+| `component`    | Declares that this is a grpc component. Value MUST be `"grpc"`. | Yes       |
 | `rpc.service`  | The service name, must be equal to the $service part in the span name. | Yes |
+| `net.peer.port`  | See [network attributes][]. | Yes |
```
Comment (bogdandrutu) [inline specification/data-rpc.md]: I am confused between this and the description of the PR. As you mentioned in the description the port is not available on the server side, so why is this required then?

File: specification/data-rpc.md [hunk 5]
```diff
@@ -32,10 +32,11 @@ Examples of span name: `grpc.test.EchoService/Echo`.
 
 | Attribute name | Notes and examples                                           | Required? |
 | -------------- | ------------------------------------------------------------ | --------- |
-| `component`    | Declares that this is a grpc component. Value MUST be `"grpc"` | Yes       |
+| `component`    | Declares that this is a grpc component. Value MUST be `"grpc"`. | Yes       |
 | `rpc.service`  | The service name, must be equal to the $service part in the span name. | Yes |
```
Comment (bogdandrutu) [inline specification/data-rpc.md]: Why do we need to duplicate this info?

File: specification/data-rpc.md (hunk 4 above)
Comment (arminru) [reply specification/data-rpc.md]: The hostname of the peer would usually not be available but the remote port should be there.

File: specification/data-rpc.md (hunk 5 above)
Comment (arminru) [reply specification/data-rpc.md]: The service name is one of the most interesting attributes for grouping and filtering on the backend so I would make that one available without having to parse the span name. I deem the attributes defined here more stable and reliable than the span name and would prefer those for analyzing.
This is, however, not part of my changeset but was already there before my PR.

Comment (Oberon00) [reply specification/data-rpc.md]: It was renamed from `peer.name` in #349.

File: specification/data-rpc.md (hunk 4 above)
Comment (bogdandrutu) [reply specification/data-rpc.md]: I still am confused, when you open a socket to a remote location usually it get allocated a port that is not easy to associate with the application, but I think you always have the ip of the peer.

Comment (arminru) [reply specification/data-rpc.md]: `net.peer.*` describes the remote side of the connection. As a client, this would be the server I'm connecting to; as a server, this would be the client connecting to me. In both cases I should be able to know both the address as well as the port that I'm connecting to, otherwise I wouldn't be able to send a message or respond.

In GRPC for Java, for example, the client knows the server's address and port since it has to be provided to [ManagedChannelBuilder.forAddress(host, port)](https://grpc.github.io/grpc-java/javadoc/io/grpc/ManagedChannelBuilder.html#forAddress-java.lang.String-int-) and the server can access the same for the client via [serverCall.getAttributes().get(Grpc.TRANSPORT_ATTR_REMOTE_ADDR)](https://grpc.github.io/grpc-java/javadoc/io/grpc/Grpc.html#TRANSPORT_ATTR_REMOTE_ADDR), which returns a `java.net.InetSocketAddress` containing the client's `InetAddress` and port.

Comment (bogdandrutu) [reply specification/data-rpc.md]: but the port on the client is useless, and changes between connections from same peers. Why is this required in that case?

Comment (arminru) [reply specification/data-rpc.md]: It's not useless if you have infrastructure monitoring that tracks open TCP connections on a host when the process itself is not instrumented.

Comment (arminru) [reply specification/data-rpc.md]: updated as discussed in the SIG meeting

File: specification/data-rpc.md [hunk 6]
```diff
@@ -32,10 +32,15 @@ Examples of span name: `grpc.test.EchoService/Echo`.
...
+| `net.peer.ip`  | See [network attributes][]. | See below |
+| `net.peer.name`  | See [network attributes][]. | See below |
+| `net.peer.port`  | See [network attributes][]. | See below |
 
-Additionally, the `net.peer.name` and `net.peer.port` [network attributes][] are required.
+At least one of [network attributes][] `net.peer.name` or `net.peer.ip` is required.
+For client-side spans describing the server they are connecting to, `net.peer.port` is required.
```
Comment (Oberon00) [inline specification/data-rpc.md]:
```suggestion
For client-side spans `net.peer.port` is required (it describes the server port they are connecting to).
```

File: specification/data-rpc.md [hunk 7]
```diff
@@ -32,10 +32,15 @@ Examples of span name: `grpc.test.EchoService/Echo`.
...
+For server-side spans describing the client connecting to them, `net.peer.port` is optional.
```
Comment (Oberon00) [inline specification/data-rpc.md]:
```suggestion
For server-side spans `net.peer.port` is optional (it describes the port the client is connecting from).
```
